### PR TITLE
fix exceptions when processing conference papers

### LINKED ISSUE #77: to_apa_citation raises error on nil during pub_hash[:conference] processing
While investigating #23, I'm seeing a `NoMethodError` exception on production data when calling `Publication.rebuild_pub_hash`, on this line:

```
cit_data_hash['event-place'] << pub_hash[:conference][:statecountry] unless pub_hash[:conference][:statecountry].blank?
```

this exception happens on a variety of publications (these are `publications.id` values):

```
22062 22388 23978 30146 37001 37531 37532 40437 47572 47573 58972
58973 65980 66980 68442 69643 70651 75863 129978 130164 130305
131242 131270 157519 159693 159694 165886 168603 169038 207992
232847 233685 238371 241848 245585 246339 248473 248717 248750
249970 251667 251686 260384 260562 260902 260905 261912 263188
266801 267460 268528 268725 268850 269040 271002 271072 271256
272814 276902 282794 282843 283394 284028 284092 284275 284679
284732 284750 286015 307497 308208 309422 312110 314590 315342
317297 318167 326246 326427 326453 330558 330559 347636 350973
351310 352634 353277 353643 354323 354324 354325 354326 354784
359388 361965 362052 363401 363859 363860 363866 364430 366205
366214 368294 368381 369770 370200 371987 372175 372674 374539
374698 374845 376218 380027 381078 385592 386907 387117 390073
394382 394903 394904 395500 395507 397695 398994 399221
```

Here's the full exception
```
ERROR 2016-06-07 12:26:41 -0700[31750]: Problem with pub 169038: #<NoMethodError: undefined method `<<' for nil:NilClass>
ERROR 2016-06-07 12:26:41 -0700[31750]: /Users/drh/workspace/sul_pub/app/models/pub_hash.rb:153:in `csl_doc'
/Users/drh/workspace/sul_pub/app/models/pub_hash.rb:54:in `to_apa_citation'
/Users/drh/workspace/sul_pub/app/models/publication.rb:350:in `update_formatted_citations'
/Users/drh/workspace/sul_pub/app/models/publication.rb:198:in `sync_publication_hash_and_db'
/Users/drh/workspace/sul_pub/app/models/publication.rb:211:in `rebuild_pub_hash'
script/fix_sw_procs_marked_as_article.rb:12:in `fix'
script/fix_sw_procs_marked_as_article.rb:50:in `block (2 levels) in search'
/Users/drh/.rvm/gems/ruby-2.2.2/gems/activerecord-4.2.5.2/lib/active_record/relation/batches.rb:51:in `block (2 levels) in find_each'
/Users/drh/.rvm/gems/ruby-2.2.2/gems/activerecord-4.2.5.2/lib/active_record/relation/batches.rb:51:in `each'
/Users/drh/.rvm/gems/ruby-2.2.2/gems/activerecord-4.2.5.2/lib/active_record/relation/batches.rb:51:in `block in find_each'
/Users/drh/.rvm/gems/ruby-2.2.2/gems/activerecord-4.2.5.2/lib/active_record/relation/batches.rb:124:in `find_in_batches'
/Users/drh/.rvm/gems/ruby-2.2.2/gems/activerecord-4.2.5.2/lib/active_record/relation/batches.rb:50:in `find_each'
script/fix_sw_procs_marked_as_article.rb:48:in `block in search'
/Users/drh/.rvm/gems/ruby-2.2.2/gems/parallel-1.6.0/lib/parallel.rb:403:in `call'
/Users/drh/.rvm/gems/ruby-2.2.2/gems/parallel-1.6.0/lib/parallel.rb:403:in `call_with_index'
/Users/drh/.rvm/gems/ruby-2.2.2/gems/parallel-1.6.0/lib/parallel.rb:372:in `process_incoming_jobs'
/Users/drh/.rvm/gems/ruby-2.2.2/gems/parallel-1.6.0/lib/parallel.rb:354:in `block in worker'
/Users/drh/.rvm/gems/ruby-2.2.2/gems/parallel-1.6.0/lib/parallel.rb:347:in `fork'
/Users/drh/.rvm/gems/ruby-2.2.2/gems/parallel-1.6.0/lib/parallel.rb:347:in `worker'
/Users/drh/.rvm/gems/ruby-2.2.2/gems/parallel-1.6.0/lib/parallel.rb:338:in `block in create_workers'
/Users/drh/.rvm/gems/ruby-2.2.2/gems/parallel-1.6.0/lib/parallel.rb:337:in `each'
/Users/drh/.rvm/gems/ruby-2.2.2/gems/parallel-1.6.0/lib/parallel.rb:337:in `create_workers'
/Users/drh/.rvm/gems/ruby-2.2.2/gems/parallel-1.6.0/lib/parallel.rb:296:in `work_in_processes'
/Users/drh/.rvm/gems/ruby-2.2.2/gems/parallel-1.6.0/lib/parallel.rb:229:in `map'
/Users/drh/.rvm/gems/ruby-2.2.2/gems/parallel-1.6.0/lib/parallel.rb:191:in `each'
script/fix_sw_procs_marked_as_article.rb:38:in `search'
script/fix_sw_procs_marked_as_article.rb:60:in `<top (required)>'
/Users/drh/.rvm/gems/ruby-2.2.2/gems/railties-4.2.5.2/lib/rails/commands/runner.rb:60:in `load'
/Users/drh/.rvm/gems/ruby-2.2.2/gems/railties-4.2.5.2/lib/rails/commands/runner.rb:60:in `<top (required)>'
/Users/drh/.rvm/gems/ruby-2.2.2/gems/railties-4.2.5.2/lib/rails/commands/commands_tasks.rb:123:in `require'
/Users/drh/.rvm/gems/ruby-2.2.2/gems/railties-4.2.5.2/lib/rails/commands/commands_tasks.rb:123:in `require_command!'
/Users/drh/.rvm/gems/ruby-2.2.2/gems/railties-4.2.5.2/lib/rails/commands/commands_tasks.rb:90:in `runner'
/Users/drh/.rvm/gems/ruby-2.2.2/gems/railties-4.2.5.2/lib/rails/commands/commands_tasks.rb:39:in `run_command!'
/Users/drh/.rvm/gems/ruby-2.2.2/gems/railties-4.2.5.2/lib/rails/commands.rb:17:in `<top (required)>'
script/rails:6:in `require'
script/rails:6:in `<main>'
```

### PR DESCRIPTION
**Pull request by drh-stanford**
This PR fixes #77. I've tested this on a variety of conference and non-conference papers. It's ready for review.

### CODE REVIEW

File: spec/models/pub_hash_spec.rb [hunk 1]
```diff
@@ -809,6 +809,31 @@ def expect_field_match(field)
...
+      let(:pub_hash) do
+        h = JSON.parse(create(:conference_proceeding).source_data, symbolize_names: true)
+        h[:conference][:location] = nil
+        h[:conference][:city] = nil
+        h[:conference][:statecountry] = 'California'
+        PubHash.new(h)
+      end
+      it 'creates citation data for event-place' do
+        expect(pub_hash.csl_doc).to include('event-place' => 'California')
+      end
```
Comment (darrenleeweber) [inline spec/models/pub_hash_spec.rb]: Add a spec for a case with a city and no state/country.

File: app/models/pub_hash.rb [hunk 2]
```diff
@@ -148,8 +148,8 @@ def csl_doc
         cit_data_hash['event-date'] = { 'date-parts' => [[pub_hash[:conference][:year]]] } unless pub_hash[:conference][:year].blank?
         cit_data_hash['number'] = pub_hash[:conference][:number] unless pub_hash[:conference][:number].blank?
         if !pub_hash[:conference][:city].blank? || !pub_hash[:conference][:statecountry].blank?
-          cit_data_hash['event-place'] = pub_hash[:conference][:city]
-          cit_data_hash['event-place'] << ',' unless pub_hash[:conference][:city].blank? || pub_hash[:conference][:statecountry].blank?
+          cit_data_hash['event-place'] = pub_hash[:conference][:city] || ''
+          cit_data_hash['event-place'] << ',' unless cit_data_hash['event-place'].blank?
```
Comment (darrenleeweber) [inline app/models/pub_hash.rb]: This might result in an extra `,` after the city when we have a city but no state/country.

**Comment by darrenleeweber** [comment]
Do we have time to add specs for `script/rebuild_conference_papers.rb` to check that it corrects the required pubs correctly?

**Comment by drh-stanford** [comment]
I've corrected the problem when there a city but no state and added specs.

Re: script specs. I'm not sure how we can add specs since the scriptie is destructive.  Strangely, there's no specs for `Publication.rebuild_pub_hash`, so maybe the action item is to add some there?

**Pull request merged**